python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -74,14 +119,14 @@ def define_out_of_pipeline_info(self, context_config):
         from dagster.core.system_config.objects import EnvironmentConfig
 
         dummy_environment_config = EnvironmentConfig(context=typed_context)
+        # BUG: If the context cleans up after itself (e.g. closes a db connection or similar)
```

Comment:
What about wrapping the return in a try...finally block?

Reply:
not sufficient. the key thing is that we need to execute the __exit__ block of the context manager after the notebook cells have been executed.